src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1344,12 +1344,12 @@ void TorrentHandle::pause()
     m_nativeHandle.pause();
 }
 
-void TorrentHandle::resume(bool forced)
+void TorrentHandle::resume(bool forced, bool uploadMode)
 {
     if (hasError())
         m_nativeHandle.clear_error();
     m_hasMissingFiles = false;
-    m_nativeHandle.set_upload_mode(false);
+    m_nativeHandle.set_upload_mode(uploadMode);
     m_nativeHandle.auto_managed(!forced);
```

Comment:
You should also switch the sequence of `set_upload_mode` & `auto_managed`, from libtorrent set_upload_mode doc:
>If the torrent is auto managed, it will automatically be taken out of upload mode periodically 

so you'll want to make sure it is not in auto manage mode first...

Reply:
Fixed.